### PR TITLE
refactor(coq): rename libraries to plugins 

### PR DESCRIPTION
**Pull request by Alizter**
Depends on #5784 

We rename the libraries field of the coq.theory stanza to plugins. Add some tests to make sure the deprecation messages are working/useful, and finally do some misc internal cleanup: (fixing warning messages, renaming internal things).

I won't touch "libraries" in Coq_lib since it didn't make sense for them to be called plugins there.

### CODE REVIEW

**Comment by ejgallego** [comment]
This looks Ok to me @Alizter , rebase please and I'll review and merge.

**Comment by Alizter** [comment]
@ejgallego done.

**Comment by Alizter** [comment]
@ejgallego Oh yes, I missed that. Let me get on it.

File: src/dune_rules/coq_stanza.ml [hunk 1]
```diff
@@ -23,36 +23,59 @@ let coq_syntax =
...
 
+  let select_deprecation ~plugins ~libraries =
+    match (plugins, libraries) with
+    | p, [] -> p
+    | [], ((loc, _) :: _ as ls) ->
+      User_warning.emit ~loc
+        [ Pp.text
+            "(libraries ...) is deprecated and will be removed before Coq \
+             language version 1.0, please use (plugins ...) instead."
+        ];
```
[the remark this answers is not in the document]
Comment (Alizter) [reply src/dune_rules/coq_stanza.ml]: Done.

**Comment by Alizter** [comment]
Alright, I've reused the stuff that Dune has for deprecated fields.

**Comment by Alizter** [comment]
I've also bumped the Coq lang version to 0.5 for Dune 3.4.

**Comment by ejgallego** [comment]
@Alizter test suite is failing, otherwise the PR is good!

**Comment by Alizter** [comment]
Unfortunately I wasn't able to install coqnative on my little laptop last week, so I never managed to run the test-suite. Should be fixed now.

**Pull request merged**